sdk/core/Azure.Core/src/TokenRequestContext.cs

```diff
@@ -67,6 +73,30 @@ public TokenRequestContext(string[] scopes, string? parentRequestId = default, s
             IsCaeEnabled = isCaeEnabled;
         }
 
+        /// <summary>
+        /// Creates a new TokenRequest with the specified scopes.
+        /// </summary>
+        /// <param name="scopes">The scopes required for the token.</param>
+        /// <param name="parentRequestId">The <see cref="Request.ClientRequestId"/> of the request requiring a token for authentication, if applicable.</param>
+        /// <param name="claims">Additional claims to be included in the token.</param>
+        /// <param name="tenantId">The tenant ID to be included in the token request.</param>
+        /// <param name="isCaeEnabled">Indicates whether to enable Continuous Access Evaluation (CAE) for the requested token.</param>
+        /// <param name="isProofOfPossessionEnabled">Indicates whether to enable Proof of Possession (PoP) for the requested token.</param>
+        /// <param name="proofOfPossessionNonce">The nonce value required for PoP token requests.</param>
+        /// <param name="request">The request to be authorized with a PoP token.</param>
+        public TokenRequestContext(string[] scopes, string? parentRequestId = default, string? claims = default, string? tenantId = default, bool isCaeEnabled = false, bool isProofOfPossessionEnabled = false, string? proofOfPossessionNonce = default, Request? request = default)
+        {
+            Scopes = scopes;
+            ParentRequestId = parentRequestId;
+            Claims = claims;
+            TenantId = tenantId;
+            IsCaeEnabled = isCaeEnabled;
+            ProofOfPossessionNonce = proofOfPossessionNonce;
+            IsProofOfPossessionEnabled = isProofOfPossessionEnabled;
+            _uri = request?.Uri.ToUri();
```

Comment:
Curious if you need the whole request passed in or if it would be better to take Uri and Method separately, even as primitives.  That might generalize better, and create fewer dependencies on other Azure.Core types going forward.

Reply:
I don't feel strongly, but I did this as a convenience. Do you foresee this being an issue for SCM request types?

Reply:
Is the convenience to have fewer parameters, or because you know of more values from Request that you might want to use in the future?  In general, using primitives in method signatures makes APIs more evolvable -- I don't have a specific SCM consideration offhand since we are still considering how to put a TokenCredential equivalent in SCM, but this has the feel of APIs that make moving things in SCM a little more complicated.

Reply:
Conversely, if we ever do end up needing more values from request, this approach would avoid having to add yet another ctor.

Reply:
Indeed.  Do you think that is a feasible scenario?  This approach adds a dependency to a new type instead of keeping it in an exclusively primitive space.  It's easy enough to add a new constructor that takes new primitives.

Reply:
Is there a specific API evolution you have in mind that would be problematic due to a dependency on a foundational Core type like `Request`?

Reply:
Sure - if we wanted to use Azure.Core types for auth, but SCM types for the pipeline, like we do today in Azure OAI client.

Reply:
OK - I've removed `Request` from the API.